### PR TITLE
Check if rangeOfString found something

### PR DESCRIPTION
**Pull request by constantintin**

### CODE REVIEW

File: libPhoneNumber/NBAsYouTypeFormatter.m [hunk 1]
```diff
@@ -810,10 +810,12 @@ - (BOOL)ableToExtractLongerNdd_ {
     NSString *prefixBeforeNationalNumberStr = [self.prefixBeforeNationalNumber_ copy];
     NSRange lastRange = [prefixBeforeNationalNumberStr rangeOfString:self.nationalPrefixExtracted_
                                                              options:NSBackwardsSearch];
-    /** @type {number} */
-    NSUInteger indexOfPreviousNdd = lastRange.location;
-    self.prefixBeforeNationalNumber_ = [[prefixBeforeNationalNumberStr
-        substringWithRange:NSMakeRange(0, indexOfPreviousNdd)] mutableCopy];
+    if (lastRange.length > 0) {
```
Comment (pres) [inline libPhoneNumber/NBAsYouTypeFormatter.m:813]: @constantintin in theory you can have range with a location and a zero length, so normally a valid range is checked by comparing the location to `NSNotFound`, e.g. here `lastRange.location != NSNotFound`

Comment (constantintin) [reply libPhoneNumber/NBAsYouTypeFormatter.m:813]: I was going off of this comment (`These methods perform string search, looking for the searchString within the receiver string.  These return length==0 if the target string is not found. So, to check for containment: ([str rangeOfString:@"target"].length > 0).  Note that the length of the range returned by these methods might be different than the length of the target string, due composed characters and such.`) found in the h-file for `-[NSString rangeOfString:]`.

Comment (constantintin) [reply libPhoneNumber/NBAsYouTypeFormatter.m:813]: Should I change this? NSNotFound makes sense to me as well.

Comment (pres) [reply libPhoneNumber/NBAsYouTypeFormatter.m:813]: It shouldn't make a difference in this case.

Comment (constantintin) [reply libPhoneNumber/NBAsYouTypeFormatter.m:813]: I'll keep `NSNotFound` in mind, TY.

**Pull request merged**